exec/java-exec/src/main/resources/rest/alertModals.ftl

```diff
@@ -0,0 +1,111 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<style>
+  .modalHeaderAlert, .closeX {
+    color:red !important;
+    text-align: center;
+    font-size: 16px;
+  }
+</style>
+  <!--
+    Alert Modal to use across templates.
+    By default, modal is hidden and expected to be populated and activated by relevant JavaScripts
+  -->
+  <div class="modal" id="errorModal" role="dialog" data-backdrop="static" data-keyboard="false" style="display: none;" aria-hidden="true">
+    <div class="modal-dialog">
+      <!-- Modal content-->
+      <div class="modal-content">
+        <div class="modal-header modalHeaderAlert">
+          <button type="button" class="close closeX" data-dismiss="modal" style="color:red;font-size:200%">×</button>
+          <h4 class="modal-title"><span class="glyphicon glyphicon-alert" style="font-size:125%"></span><span id="modalHeader" style="font-family:Helvetica Neue,Helvetica,Arial,sans-serif;white-space:pre">~ErrorMessage~ Title</span></h4>
+        </div>
+        <div class="modal-body" id="modalBody" style="line-height:3">
+        ~ErrorMessage Details~
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-info btn-default" data-dismiss="modal">Close</button>
+        </div>
+      </div>
+    </div>
+  </div>
+<script>
+    //Populate the alert modal with the right message params and show
+    function populateAndShowAlert(errorMsg, inputValues) {
+      //console.log("placeHolder -> "+ JSON.stringify(errorMap[errorMsg].placeHolder));
+      //console.log("inputValues -> "+ JSON.stringify(inputValues));
+      modalHeader.innerHTML=errorMap[errorMsg].msgHeader;
+      modalBody.innerHTML=errorMap[errorMsg].msgBody;
+      let substNames=[];
+      let substValues=[];
+      //Check if substitutions are needed
+      if (errorMap[errorMsg].placeHolder != null) {
+        //Check if input is an array of multiple substitutions
+        if (Array.isArray(errorMap[errorMsg].placeHolder)) {
+          substNames=errorMap[errorMsg].placeHolder;
+          substValues=inputValues;
+        } else {
+            substNames=[errorMap[errorMsg].placeHolder];
+            substValues=[inputValues];
+        }
+        let updatedHtml=modalBody.innerHTML;
+        for (i=0; i<substNames.length && i<substValues.length; i++) {
+            console.log(substNames[i] + " => " + substValues[i]);
+            updatedHtml=updatedHtml.replace(substNames[i], substValues[i]);
+        }
+        modalBody.innerHTML=updatedHtml;
+      }
+      //Show Alert
+      $('#errorModal').modal('show');
```

Comment:
is there a way to put the mouse focus on erroneous field so that user can make the correction ? If not then I will avoid using modal for login page.

Reply:
It is possible but will be non-trivial work, as I'll need to pass an additional field to indicate which field to focus on.

Reply:
Done